app/controllers/spree/admin/payment_methods_controller.rb

```diff
@@ -34,16 +32,16 @@ def update
         force_environment
 
         invoke_callbacks(:update, :before)
-        payment_method_type = params[:payment_method].delete(:type)
-        if @payment_method['type'].to_s != payment_method_type
+
+        if @payment_method['type'].to_s != payment_method_class
           @payment_method.update_columns(
-            type: payment_method_type,
+            type: payment_method_class,
```

Comment:
given the result of the earlier call isn't memorized, isn't this `nil`? The second time `base_params` no longer contains the `:type` key.

Reply:
Woah, nice catch.

Reply:
Fixed, and test coverage added :+1: